src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1,79 +1,1419 @@
+          }
+          sqlFilterExpression += `id ${doInclude ? '=' : '!='} '${jurId}'`;
+        }
+      }
+
+      const otherJurisdictionSupersetParams = sqlFilterExpression.length
+        ? superset.getFormData(3000, [{ sqlExpression: sqlFilterExpression }])
+        : { row_limit: 3000 };
+
+      supersetService(SUPERSET_JURISDICTIONS_DATA_SLICE, otherJurisdictionSupersetParams).then(
+        (jurisdictionResults: FlexObject[] = []) => {
+          const jurisdictions = jurisdictionResults.map(j => {
+            const { id, parent_id, name, geographic_level } = j;
+            const jurisdiction: Jurisdiction = {
+              geographic_level: geographic_level || 0,
+              jurisdiction_id: id,
+              name: name || null,
+              parent_id: parent_id || null,
+            };
+            return jurisdiction;
+          });
+          return fetchJurisdictionsActionCreator(jurisdictions);
+        }
+      );
+    }
+  }
+
+  public componentWillReceiveProps(nextProps: IrsPlanProps) {
+    const {
+      childlessChildrenIds,
+      country,
+      isLoadingGeoms,
+      isLoadingJurisdictions,
+      newPlan,
+    } = this.state;
+    const { isDraftPlan, jurisdictionsArray, planById } = nextProps;
+
+    if (newPlan && childlessChildrenIds && country && isLoadingGeoms) {
+      const filteredJurisdictions = jurisdictionsArray.filter(
+        (j: Jurisdiction) => childlessChildrenIds.indexOf(j.jurisdiction_id) !== -1
+      );
+
+      const loadedJurisdictions = filteredJurisdictions.filter((j: Jurisdiction) => j.geojson);
+      if (loadedJurisdictions.length === filteredJurisdictions.length) {
+        this.setState({
+          isLoadingGeoms: false,
+        });
+      }
+    }
+
+    if (
+      isLoadingJurisdictions &&
+      jurisdictionsArray.length !== this.props.jurisdictionsArray.length
+    ) {
+      this.setState({ isLoadingJurisdictions: false });
+    }
+
+    if (isDraftPlan && !newPlan && planById && planById.plan_jurisdictions_ids) {
+      this.setState({
+        isStartingPlan: !!this.state.country,
+        newPlan: planById,
+      });
+    }
   }
 
   public render() {
-    const { planId, planById, isDraftPlan, isFinalizedPlan } = this.props;
-    if (planId && !planById) {
+    const { planId, planById, isDraftPlan, isFinalizedPlan, isNewPlan } = this.props;
+    const {
+      planCountry,
+      doRenderTable,
+      tableCrumbs,
+      newPlan,
+      isEditingPlanName,
+      isLoadingJurisdictions,
+      isStartingPlan,
+    } = this.state;
+    if (
+      (planId && !planById) ||
+      (isNewPlan && !newPlan) ||
+      (isDraftPlan && isLoadingJurisdictions)
+    ) {
       return <Loading />;
     }
+
+    const pageLabel =
+      (isFinalizedPlan && planById && planById.plan_title) ||
+      (isDraftPlan && planById && `${planById.plan_title} (draft)`) ||
+      (newPlan && newPlan.plan_title) ||
+      'New Plan';
```

Comment:
I think we have used this string a number of times in this module.  In the spirit of zero tolerance to magic strings, should we move it to the constants module?

Reply:
Sure thing! ...what's a magic string?

Reply:
Here you go https://deviq.com/magic-strings/

Reply:
Cool deal, replace this one but seems like quite a few of my strings could be considered magical, created a new issue at #238 to audit and replace in the near future.